health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/data/ExerciseClientManager.kt

```diff
@@ -168,6 +192,9 @@ class ExerciseClientManager @Inject constructor(
      * cancelled, this flow will unregister the listener.
      * [callbackFlow] is used to bridge between a callback-based API and Kotlin flows.
      */
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @SuppressLint("MissingPermission")
```

Comment:
Worrying to have suppressions. Are there ways to structure without needing a suppression?